sys/net/gnrc/network_layer/ndp2/gnrc_ndp2.c

```diff
@@ -0,0 +1,567 @@
+                gnrc_pktbuf_release(pkt);
+                return;
+            }
+        }
+    }
+#ifdef MODULE_GNRC_SIXLOWPAN_ND
+    /* add ARO based on interface */
+    if (netif->flags & GNRC_IPV6_NETIF_FLAGS_SIXLOWPAN) {
+        if (l2src_len != sizeof(eui64_t)) {
+            l2src_len = (uint16_t)gnrc_netapi_get(netif->pid,
+                                                  NETOPT_ADDRESS_LONG, 0,
+                                                  l2src, sizeof(l2src));
+            if (l2src_len != sizeof(eui64_t)) {
+                DEBUG("ndp2: can't get EUI-64 of the interface\n");
+                gnrc_pktbuf_release(pkt);
+                return;
+            }
+        }
+        hdr = gnrc_sixlowpan_nd_opt_ar_build(0, GNRC_SIXLOWPAN_ND_AR_LTIME,
+                                             (eui64_t *)l2src, pkt);
+        if (hdr == NULL) {
+            DEBUG("ndp2: error allocating ARO.\n");
+            gnrc_pktbuf_release(pkt);
+            return;
+        }
+        pkt = hdr;
+    }
+#endif  /* MODULE_GNRC_SIXLOWPAN_ND */
+    /* add neighbor solicitation header */
+    hdr = gnrc_ndp2_nbr_sol_build(tgt, pkt);
+    if (hdr == NULL) {
+        DEBUG("ndp2: error allocating neighbor solicitation.\n");
+        gnrc_pktbuf_release(pkt);
+        return;
+    }
+    pkt = hdr;
+    /* add remaining headers */
+    hdr = _build_headers(netif, src, dst, pkt);
+    if (hdr == NULL) {
+        DEBUG("ndp2: error adding lower-layer headers.\n");
+        gnrc_pktbuf_release(pkt);
+    }
+    else if (gnrc_netapi_dispatch_send(GNRC_NETTYPE_NDP2,
+                                       GNRC_NETREG_DEMUX_CTX_ALL, hdr) == 0) {
+        DEBUG("ndp2: unable to send neighbor solicitation\n");
+        gnrc_pktbuf_release(hdr);
+    }
+}
+
+void gnrc_ndp2_nbr_adv_send(const ipv6_addr_t *tgt, gnrc_ipv6_netif_t *netif,
+                            const ipv6_addr_t *dst, bool supply_tl2a,
+                            gnrc_pktsnip_t *ext_opts)
+{
+    ipv6_addr_t real_dst;
+    gnrc_pktsnip_t *hdr, *pkt = ext_opts;
+    uint8_t adv_flags = 0;
+
+    assert((tgt != NULL) && !ipv6_addr_is_multicast(tgt));
+    assert((netif != NULL) && (dst != NULL));
+    DEBUG("ndp2: send neighbor advertisement (iface: %" PRIkernel_pid
+          ", tgt: %s, ", netif->pid,
+          ipv6_addr_to_str(addr_str, tgt, sizeof(addr_str)));
+    DEBUG("dst: %s, supply_tl2a: %d)\n",
+          ipv6_addr_to_str(addr_str, dst, sizeof(addr_str)), supply_tl2a);
+    if ((netif->flags & GNRC_IPV6_NETIF_FLAGS_ROUTER) &&
+        (netif->flags & GNRC_IPV6_NETIF_FLAGS_RTR_ADV)) {
+        adv_flags |= NDP_NBR_ADV_FLAGS_R;
+    }
+    if (ipv6_addr_is_unspecified(dst)) {
+        memcpy(&real_dst, &ipv6_addr_all_nodes_link_local, sizeof(ipv6_addr_t));
+        supply_tl2a = true;
+    }
+    else {
+        memcpy(&real_dst, dst, sizeof(real_dst));
+        adv_flags |= NDP_NBR_ADV_FLAGS_S;
+    }
+    /* add SL2AO based on target address */
```

Comment:
>(there are only two instances of "SLLAO" in RFC 4861)

and 20 instances of "SLLAO" in RFC 6775 .. so why not use the same terminology ..